### PR TITLE
WIP: Get matching field tests

### PR DESCRIPTION
**Pull request by amitaibu**

### CODE REVIEW

**Comment by amitaibu** [comment]
@damiankloip I'm wondering if unit test in the WIP PR is the right thing. The amount of mockups is pretty big - do you think it's worth the effort, or should we opt to Kernel test?

**Comment by amitaibu** [comment]
oh, meant to ping also @chx :)

**Comment by damiankloip** [comment]
@amitaibu I would still go with the unit test. That amount of mocking is nothing compared to some places in core :)

Kernel Tests can be used too, as more of a semi-integration test for more complete code paths?

**Comment by amitaibu** [comment]
@damiankloip Challenge accepted -- although I might ping you for some help ;)

File: tests/src/Unit/GetMatchingFieldTest.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    // never expected, so just check it's not called.
+    $this->groupContent->get($field_name)
+      ->shouldNotBeCalled();
+
+    $this->entityType = $this->getMock('\Drupal\Core\Entity\EntityTypeInterface');
+    $this->entityType->expects($this->any())
+      ->method('getProvider')
+      ->will($this->returnValue($this->provider));
+
+    $this->entityManager = $this->getMock('\Drupal\Core\Entity\EntityTypeManagerInterface');
```
Comment (amitaibu) [inline tests/src/Unit/GetMatchingFieldTest.php]: @pfrenssen maybe you will notice what am I'm doing wrong here with the mocking?

When running test, `./vendor/phpunit/phpunit/phpunit  -c ./core/phpunit.xml.dist ./modules/og/tests/src/Unit/GetMatchingFieldTest.php`

I get:

```
PHPUnit 4.8.11 by Sebastian Bergmann and contributors.

F

Time: 260 ms, Memory: 7.75Mb

There was 1 failure:

1) Drupal\Tests\og\Unit\GetMatchingFieldTest::testGetMatchingField
Expectation failed for method name is equal to <string:getDefinition> when invoked zero or more times
Parameter 0 for invocation Drupal\Core\Entity\EntityTypeManagerInterface::getDefinition(null, true) does not match expected value.
Failed asserting that null matches expected 'tjm8jtse'.

/Library/WebServer/Documents/d8_dev/modules/og/src/OgGroupAudienceHelper.php:152
/Library/WebServer/Documents/d8_dev/modules/og/src/OgGroupAudienceHelper.php:94
/Library/WebServer/Documents/d8_dev/modules/og/tests/src/Unit/GetMatchingFieldTest.php:127
```

Unit testing isn't trivial...

Comment (chx) [inline tests/src/Unit/GetMatchingFieldTest.php]: Why are we still mocking instead of prophesizing?

Comment (amitaibu) [inline tests/src/Unit/GetMatchingFieldTest.php]: because I didn't know better :)

I'm mostly copying from core, where I see this pattern being used.

Comment (pfrenssen) [inline tests/src/Unit/GetMatchingFieldTest.php]: I agree, it's a bit strange to mix them up. Prophecy is easier to use. It does have its own limitations though.

Maybe try to prophesize it for now? If that doesn't solve it I'd be happy to dive in.

Comment (amitaibu) [inline tests/src/Unit/GetMatchingFieldTest.php]: weirdly enough, there is only a single `$container->set('entity_type.manager', $this->entityTypeManager->reveal());` in core (I see still lots of `$container->set('entity.manager'`)

Comment (amitaibu) [inline tests/src/Unit/GetMatchingFieldTest.php]: Sure, I'll try to prophesize

Comment (chx) [inline tests/src/Unit/GetMatchingFieldTest.php]: I have left enough prophecies in Og itself to write an even newer testament ;)

Comment (amitaibu) [inline tests/src/Unit/GetMatchingFieldTest.php]: I'm jewish, so I believe only on the old testament 😉 

I've started following. So far so good...

**Comment by amitaibu** [comment]
Apart of the tests not being complete, the functionality of `OgGroupAudienceHelper::getMatchingField` is not used. I'll need to revisit it.

Now that we have a distinction between group content and users, lets see how much code we can save + have a nicer DX.

**Comment by amitaibu** [comment]
This will not be needed if #265 goes in.

**Comment by amitaibu** [comment]
Not needed any more.

**Pull request closed without merging**